### PR TITLE
ZK-4016: HtmlMacroComponent doesn't listen to onCreate event with @Li…

### PR DESCRIPTION
**Pull request by klyvechen**
…sten

### CODE REVIEW

File: zktest/src/archive/test2/B86-ZK-4016.zul [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+                
+        History:
+                Tue Aug 07 14:42:46 CST 2018, Created by klyve
+
+Copyright (C) 2018 Potix Corporation. All Rights Reserved.
+
+-->
+<?component name="mylistbox" macroURI="B86-ZK-4016-mylistbox.zul"
+	class="org.zkoss.zktest.test2.B86_ZK_4016HMComponent"?>
+<zk>
```
[the remark this answers is not in the document]
Comment (klyvechen) [reply zktest/src/archive/test2/B86-ZK-4016.zul]: sorry

File: zktest/src/org/zkoss/zktest/test2/B86_ZK_4016HMComponent.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+package org.zkoss.zktest.test2;
+
+import org.zkoss.zk.ui.HtmlMacroComponent;
+import org.zkoss.zk.ui.event.Event;
+import org.zkoss.zk.ui.select.annotation.Listen;
+
+public class B86_ZK_4016HMComponent extends HtmlMacroComponent {
+	@Listen ("onCreate = #box")
+	public void create(Event e){ 
+		B86_ZK_4016FileDealer.writeMsg(this.getClass() + e.getName());
```
[the remark this answers is not in the document]
Comment (klyvechen) [reply zktest/src/org/zkoss/zktest/test2/B86_ZK_4016HMComponent.java]: ok

File: zktest/src/archive/test2/B86-ZK-4016.zul (hunk 1 above)
[the remark this answers is not in the document]
Comment (klyvechen) [reply zktest/src/archive/test2/B86-ZK-4016.zul]: sorry, ok

File: zktest/src/org/zkoss/zktest/test2/B86_ZK_4016Composer.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+*/
+package org.zkoss.zktest.test2;
+
+import org.zkoss.zk.ui.select.SelectorComposer;
+import org.zkoss.zk.ui.select.annotation.Listen;
+
+public class B86_ZK_4016Composer extends SelectorComposer {
+	@Listen ("onClick = listitem")
+	public void click(){
+		B86_ZK_4016FileDealer.writeMsg("click");
```
[the remark this answers is not in the document]
Comment (klyvechen) [reply zktest/src/org/zkoss/zktest/test2/B86_ZK_4016Composer.java]: ok

File: zktest/src/archive/test2/config.properties [hunk 4]
```diff
@@ -2511,6 +2511,7 @@ B85-ZK-4001.zul=A,E,Borderlayout,title,themes
 ## B86
 ##ztl##B86-ZK-4008.zul=A,E,Column,flex,width
 ##zats##B86-ZK-3996.zul=A,E,Fragment,lexical
+##zats##B86-ZK-4016.zul=A,E,HtmlMicroComponent,@Listener,onCreate
```
[the remark this answers is not in the document]
Comment (klyvechen) [reply zktest/src/archive/test2/config.properties]: ok

File: zktest/test/java/org/zkoss/zktest/zats/test2/B86_ZK_4016Test.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+import org.junit.Assert;
+import org.junit.Test;
+
+import org.zkoss.zktest.zats.WebDriverTestCase;
+
+public class B86_ZK_4016Test extends WebDriverTestCase {
+
+	@Test
+	public void test() {
+		B86_ZK_4016FileDealer.writeMsg("", false);
```
Comment (scribetw) [inline zktest/test/java/org/zkoss/zktest/zats/test2/B86_ZK_4016Test.java]: Remove

File: zktest/test/java/org/zkoss/zktest/zats/test2/B86_ZK_4016Test.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+		connect();
+		waitResponse();
+		click(jq("$box").find("@listitem").eq(0));
+		waitResponse();
+		Assert.assertEquals("class org.zkoss.zktest.test2.B86_ZK_4016HMComponentonCreate\nclick\n", jq("#zk_log").toElement().get("value"));
```
Comment (scribetw) [inline zktest/test/java/org/zkoss/zktest/zats/test2/B86_ZK_4016Test.java]: getZKLog()

**Comment by scribetw** [comment]
Will be re-implement by James

**Pull request closed without merging**